sailfishusermanager.cpp

```diff
@@ -309,6 +326,50 @@ int SailfishUserManager::removeUserFiles(uint uid)
     return rv;
 }
 
+/*
+ * Sets user quota limits, if supported by kernel and enabled on /home filesystem
+ */
+void SailfishUserManager::setUserLimits(uint uid)
+{
+    struct statvfs info;
+    errno = 0;
+    if (statvfs("/home", &info) < 0) {
```

Comment:
once bitten etc... Suitable libc/kernel mismatch can lead to statvfs() and like filling in only part of info buffer, leaving the rest to whatever happened to be in stack and return success. This can then lead to really confusing and hard to figure out situations -> I've made it a habit to clear such buffers before use.

Reply:
I wonder if I should also check the value but I guess the code "conveniently" sets the limit to 0 (i.e. not set) if it fails now. Then again it does that silently so maybe not so great?

Reply:
In this case: the sizes and counts have probably always been there -> does not really matter -> feel free to leave it as-is.

Also normally kernel and libc can be assumed to be in sync about buffer sizes and content.

But to avoid spending time e.g. wondering why zypper dup in sbox ends up having every 37th or so package failing to install because target file system is supposedly read-only [due to stack noise in f_flag] -> I just zero-initialize these things ;-)